### PR TITLE
Add some common generics helpers

### PR DESCRIPTION
**Pull request by Sypheos**
### What does this PR do?

Add some generics helpers that are useful in common situation, such a building a filter out of a slice.

### What are the observable changes?

New helpers for uniqueness in slice
New helpers for ptr and slice uniqueness

`gofmt -w -r 'interface{} -> any' *.go`

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Properly labeled

### CODE REVIEW

**Comment by AlexisMontagne** [comment]
Some of our project still run in older go version, you can not break all the applications just like that. 

A way to go about it would be to create new files that are  go1.18 or above only

**Comment by Sypheos** [comment]
You mean to revert the last commit and leave the `interfaces{}` or smth else ?

File: httputil/mux.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ func NewMux() *http.ServeMux {
 		mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
 		mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
 		mux.HandleFunc("/debug/pprof/heap", pprof.Index)
+		mux.Handle("/debug/pprof/", http.HandlerFunc(pprof.Index))
```
Comment (AlexisMontagne) [inline httputil/mux.go:19]: Is that related to this PR?

File: generics/unique.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
+//go:build go1.18
+
+package generics
+
+// Unique creates a map with keys being the values of a slice
+func Unique[T comparable](s []T) map[T]struct{} {
```
Comment (AlexisMontagne) [inline generics/unique.go:6]: should we call it `Set` instead? To follow the convention we created?

File: httputil/mux.go (hunk 1 above)
Comment (Sypheos) [reply httputil/mux.go:19]: no, should remove it

File: generics/unique.go (hunk 2 above)
Comment (jlevesy) [reply generics/unique.go:6]: My two cents, Should be `NewSet` then, you're creating a Set from a collection.

Though, `Unique` or `Uniq` (ruby <3) is fine.

**Review by AlexisMontagne — APPROVED** [review]

**Review by jlevesy — APPROVED** [review]
